InteractionBase/src/commands/CommandExecutionEngine.cpp

```diff
@@ -188,7 +188,9 @@ QList<CommandSuggestion*> CommandExecutionEngine::suggestionsForHandler(GenericH
 			if (alreadySuggested.contains(typeid(*command).hash_code())) continue;
 
 			QList<CommandSuggestion*> suggestions;
-			if (!trimmedCommandText.isEmpty() || command->appearsInMenus())
+			//If it is a menu command, we must make sure it can be interpreted on the current item in its current state
+			if (!trimmedCommandText.isEmpty() ||
+					(command->appearsInMenus() && command->canInterpret(source, target, QStringList(command->name()), cursor)))
```

Comment:
So this is not quite right. The idea here is that we always call suggest, and that you return 0 or more suggestions if you can handle the input text in some way. So instead of call canInterpret in addition, you should just modify your command's suggest to return an empty list in cases where they are not applicable.

Your current solution prevents a command from seeing some piece of text (e.g. "please compile the program") and suggesting a "fix" for it (e.g. suggesting the "compile" command, based on the text above).

In general, one can assume that canInterpret is true inside the body of the execute method of a command, but not insdie the body of the suggest() method.